libs/pika/async_cuda/include/pika/async_cuda/then_on_host.hpp

```diff
@@ -194,22 +236,23 @@ namespace pika::cuda::experimental {
             friend cuda_scheduler tag_invoke(
                 pika::execution::experimental::get_completion_scheduler_t<
                     pika::execution::experimental::set_value_t>,
-                then_on_host_sender const& s)
+                then_on_host_sender_type const& s) noexcept
             {
                 return s.sched;
             }
         };
 
+        // TODO: Uncomment again.
         // This should be a hidden friend in then_on_host_receiver. However,
         // nvcc does not know how to compile it with some argument types
         // ("error: no instance of overloaded function std::forward matches the
         // argument list").
-        template <typename Receiver, typename F, typename... Ts>
-        void tag_invoke(pika::execution::experimental::set_value_t,
-            then_on_host_receiver<Receiver, F>&& receiver, Ts&&... ts)
-        {
-            receiver.set_value(PIKA_FORWARD(Ts, ts)...);
-        }
+        //template <typename Receiver, typename F, typename... Ts>
+        //void tag_invoke(pika::execution::experimental::set_value_t,
+        //    then_on_host_receiver<Receiver, F>&& receiver, Ts&&... ts)
+        //{
+        //    PIKA_MOVE(receiver).set_value(PIKA_FORWARD(Ts, ts)...);
+        //}
```

Comment:
This is related to the previous `tag_invoke(set_value_t, ...)` overload comment. Investigate with nvcc.

Reply:
See https://github.com/pika-org/pika/pull/330#discussion_r926753856.